### ISSUE TITLE
[Usage] RuntimeError: probability tensor contains either `inf`, `nan` or element < 0

### ISSUE THREAD

**Issue by xuweidongkobe** [haotian-liu/LLaVA#896]
### Describe the issue

Issue:
When I run the run_llava.py，encountered this error.
I have tried other solutions mentioned by the issue, including using single card inference, fp32, commentet "do_sample", but still can not fix.
1、I am using the Nvidia v100 gpu.
2、When I comment "do_sample", code can not occur error, but the outputs is None. Its very strange.

Command:
```
PASTE THE COMMANDS HERE.
```

Log: 
```
import os 
os.environ["CUDA_VISIBLE_DEVICES"] = "6"
import argparse
import torch
from llava.constants import IMAGE_TOKEN_INDEX, DEFAULT_IMAGE_TOKEN, DEFAULT_IM_START_TOKEN, DEFAULT_IM_END_TOKEN
from llava.conversation import conv_templates, SeparatorStyle
from llava.model.builder import load_pretrained_model
from llava.utils import disable_torch_init
from llava.mm_utils import tokenizer_image_token, get_model_name_from_path, KeywordsStoppingCriteria

from PIL import Image

import requests
from PIL import Image
from io import BytesIO


def load_image(image_file):
    if image_file.startswith('http') or image_file.startswith('https'):
        response = requests.get(image_file)
        image = Image.open(BytesIO(response.content)).convert('RGB')
    else:
        image = Image.open(image_file).convert('RGB')
    return image


def eval_model(args):
    # Model
    disable_torch_init()

    model_name = get_model_name_from_path(args.model_path)
    tokenizer, model, image_processor, context_len = load_pretrained_model(args.model_path, args.model_base, model_name)

    qs = args.query
    if model.config.mm_use_im_start_end:
        qs = DEFAULT_IM_START_TOKEN + DEFAULT_IMAGE_TOKEN + DEFAULT_IM_END_TOKEN + '\n' + qs
    else:
        qs = DEFAULT_IMAGE_TOKEN + '\n' + qs

    if 'llama-2' in model_name.lower():
        conv_mode = "llava_llama_2"
    elif "v1" in model_name.lower():
        conv_mode = "llava_v1"
    elif "mpt" in model_name.lower():
        conv_mode = "mpt"
    else:
        conv_mode = "llava_v0"

    if args.conv_mode is not None and conv_mode != args.conv_mode:
        print('[WARNING] the auto inferred conversation mode is {}, while `--conv-mode` is {}, using {}'.format(conv_mode, args.conv_mode, args.conv_mode))
    else:
        args.conv_mode = conv_mode

    print("conv_mode is :", conv_mode)
    conv = conv_templates[args.conv_mode].copy()
    conv.append_message(conv.roles[0], qs)
    conv.append_message(conv.roles[1], None)
    prompt = conv.get_prompt()

    image = load_image(args.image_file)
    image_tensor = image_processor.preprocess(image, return_tensors='pt')['pixel_values'].half().cuda()
    # image_tensor = image_processor.preprocess(image, return_tensors='pt')['pixel_values'].cuda()
    input_ids = tokenizer_image_token(prompt, tokenizer, IMAGE_TOKEN_INDEX, return_tensors='pt').unsqueeze(0).cuda()

    stop_str = conv.sep if conv.sep_style != SeparatorStyle.TWO else conv.sep2
    keywords = [stop_str]
    stopping_criteria = KeywordsStoppingCriteria(keywords, tokenizer, input_ids)
    
    # model.float()
    with torch.inference_mode():
        output_ids = model.generate(
            input_ids,
            images=image_tensor,
            do_sample=True,
            temperature=0.2,
            max_new_tokens=1024,
            use_cache=True,
            stopping_criteria=[stopping_criteria])

    input_token_len = input_ids.shape[1]
    n_diff_input_output = (input_ids != output_ids[:, :input_token_len]).sum().item()
    if n_diff_input_output > 0:
        print(f'[Warning] {n_diff_input_output} output_ids are not the same as the input_ids')
    outputs = tokenizer.batch_decode(output_ids[:, input_token_len:], skip_special_tokens=True)[0]
    outputs = outputs.strip()
    if outputs.endswith(stop_str):
        outputs = outputs[:-len(stop_str)]
    outputs = outputs.strip()
    print(outputs)

if __name__ == "__main__":
    parser = argparse.ArgumentParser()
    parser.add_argument("--model-path", type=str, default="/mnt/data1/home/xuwd/code/LLaVA-1.1.3/checkpoints/IVA/1201_llava-v1.5-7b-lora/")
    parser.add_argument("--model-base", type=str, default="/mnt/data1/home/xuwd/checkpoints/LLaVA/llava-v1.5-7b/")
    parser.add_argument("--image-file", type=str, default="/mnt/data1/home/xuwd/datasets/IVA/test_data/aqmjc/helm_000000.jpg")
    parser.add_argument("--query", type=str, default="Please describe this image.")
    parser.add_argument("--conv-mode", type=str, default=None)
    args = parser.parse_args()

    eval_model(args)
```

Screenshots:
![image](https://github.com/haotian-liu/LLaVA/assets/25764883/9cd3303f-36ba-46ef-af14-da19f4cb93ac)

**Comment by xuweidongkobe** [comment]
@haotian-liu can you help me , thank you!

**Comment by xuweidongkobe** [comment]
fix the probelm with 1.1.1tag code.

**Comment by Zhou-Hangyu** [comment]
> fix the probelm with 1.1.1tag code.

Do you mean you switch back to llava1.1.1 and solved the issue?

**Comment by xuweidongkobe** [comment]
> > fix the probelm with 1.1.1tag code.
> 
> Do you mean you switch back to llava1.1.1 and solved the issue?

yes

**Comment by 459737087** [comment]
but why @xuweidongkobe

**Comment by Saurabhbaghel** [comment]
> fix the probelm with 1.1.1tag code.
@xuweidongkobe , did you make the change in `model-base` ? 
Thanks
